src/components/Pressable/PressableWithDelayToggle.tsx

```diff
@@ -1,132 +1,129 @@
      */
-    inline: PropTypes.bool,
-};
-
-const defaultProps = {
-    text: '',
-    textChecked: '',
-    tooltipText: '',
-    tooltipTextChecked: '',
-    styles: [],
-    textStyles: [],
-    iconStyles: [],
-    icon: null,
-    inline: true,
-    iconChecked: Expensicons.Checkmark,
-    innerRef: () => {},
+    inline?: boolean;
 };
 
-function PressableWithDelayToggle(props) {
+function PressableWithDelayToggle(
+    {
+        iconChecked = Expensicons.Checkmark,
+        inline = true,
+        onPress,
+        text,
+        textChecked,
+        tooltipText,
+        tooltipTextChecked,
+        styles: pressableStyle,
+        textStyles,
+        iconStyles,
+        icon,
+    }: PressableWithDelayToggleProps,
+    ref: ForwardedRef<RNText | View>,
+) {
     const [isActive, temporarilyDisableInteractions] = useThrottledButtonState();
 
     const updatePressState = () => {
         if (!isActive) {
             return;
         }
         temporarilyDisableInteractions();
-        props.onPress();
+        onPress();
     };
 
     // Due to limitations in RN regarding the vertical text alignment of non-Text elements,
     // for elements that are supposed to be inline, we need to use a Text element instead
     // of a Pressable
-    const PressableView = props.inline ? Text : PressableWithoutFeedback;
-    const tooltipText = !isActive ? props.tooltipTextChecked : props.tooltipText;
+    const PressableView = inline ? Text : PressableWithoutFeedback;
+    const tooltipTexts = !isActive ? tooltipTextChecked : tooltipText;
     const labelText = (
         <Text
             suppressHighlighting
-            style={props.textStyles}
+            style={textStyles}
         >
-            {!isActive && props.textChecked ? props.textChecked : props.text}
+            {!isActive && textChecked ? textChecked : text}
             &nbsp;
         </Text>
     );
 
     return (
         <PressableView
-            ref={props.innerRef}
+            // Using `ref as any` due to variable component (Text or View) based on 'inline' prop; TypeScript workaround.
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            ref={ref as any}
             onPress={updatePressState}
-            accessibilityLabel={tooltipText}
-            suppressHighlighting={props.inline ? true : undefined}
+            accessibilityLabel={tooltipTexts}
+            suppressHighlighting={inline ? true : undefined}
         >
             <>
-                {props.inline && labelText}
+                {inline && labelText}
                 <Tooltip
                     containerStyles={[styles.flexRow]}
-                    text={tooltipText}
+                    text={tooltipTexts}
+                    shouldRender
                 >
                     <PressableWithoutFeedback
                         focusable={false}
                         accessible={false}
+                        accessibilityLabel=""
```

Comment:
I'm not sure if it is safe from the accessibility point of view to leave it empty. Maybe we should pass `tooltipTexts` same as in `PressableView`. Thanks to that react native Pressable component will get the label

Reply:
`accessibilityLabel` shouldn't be required as long as pressable is not accessible. I remember that I have implemented the check in propTypes for it

Reply:
Adjusted the type and removed `accessibilityLabel=""`, thanks!